### ISSUE TITLE
OpenPgpService - Signing Subkey Not Found

### ISSUE THREAD

**Issue by alephex** [open-keychain/open-keychain#1413]
#1402 is still a problem in version 3.4

----

### OpenPgpService

Signing within the OpenKeychain app works well.

Signing email via K-9 Mail fails with `OpenPGP error: signing subkey not found!`

The key looks like this: 

Key | Public | Secret | Size
---- | ---- | ---- | ----
Master (stripped) | Yes | No | 4096
Encrypt Subkey | Yes | Yes | 4096
Sign Subkey | Yes | Yes | 4096

OpenKeychain finds the signing subkey when used standalone. It does *not* find the subkey via OpenPgpService.

#### Logcat Exerpt
```
I/Timeline( 4158): Timeline: Activity_launch_request id:com.fsck.k9 time:1484941
I/ActivityManager( 3170): START u0 {act=com.fsck.k9.intent.action.COMPOSE cmp=com.fsck.k9/.activity.MessageCompose (has extras)} from pid 4158
I/LatinIME:LogUtils( 3411): Dictionary info: dictionary = contacts.en_GB.dict ; version = ? ; date = ?
I/ActivityManager( 3170): Displayed com.fsck.k9/.activity.MessageCompose: +255ms
I/Timeline( 4158): Timeline: Activity_idle id: android.os.BinderProxy@4221e640 time:1485357
I/Timeline( 3170): Timeline: Activity_windows_visible id: ActivityRecord{42333948 u0 com.fsck.k9/.activity.MessageCompose t8} time:1485549
W/Keychain( 4820): You are using a deprecated API call, please use ACTION_CLEARTEXT_SIGN instead of ACTION_SIGN!
E/k9      ( 4158): OpenPGP Error ID:0                                           
E/k9      ( 4158): OpenPGP Error Message:signing subkey not found!
```

@Valodim mentioned in #1402 that:
> the problem is that the (master) key id in the accounts database is zero.

Which, I assume, is this `E/k9      ( 4158): OpenPGP Error ID:0`

It also seems that K-9 Mail is using API calls which have been depreciated. Is this likely to be a factor?

----

Android 4.4.4
K-9 Mail 5.006
OpenKeychain 3.4

**Comment by Valodim** [comment]
did you re-configure this? the problem is in the database, it doesn't fix itself unfortunately :(

**Comment by dschuermann** [comment]
@alephex Please try revoking the access of K-9 Mail and then configuring it again by sending a new signed email.
Revoking can be done from OpenKeychain -> Apps -> K-9 Mail and then in the overflow menu

**Comment by alephex** [comment]
@Valodim 
I reimported the key and had the same problem.

@dschuermann 
That did the trick. Signing and verifying are working.

Thanks folks.

**Comment by Valodim** [comment]
we will have a bugfix release anyways, deleting all 0-entries sounds like a good idea there

**Comment by Valodim** [comment]
backport-fix applied in af546ae10190ac076cd7b091cd064fb8a5c03519
